src/main.rs

```diff
@@ -69,21 +75,76 @@ impl Group {
 struct DWebBackend {
     path: String,
     port: u16,
+    store: Store,
+    veilid_api: Option<VeilidAPI>,
 }
 
 impl DWebBackend {
-    pub fn new(path: String, port: u16) -> Self {
-        DWebBackend { path, port }
-    }
-    pub async fn start(&mut self) -> Result<()> {
-        // Init veilid
-        println!("Starting in {} with port {}", self.path, self.port);
+    pub fn new(base_path: &str, port: u16) -> Self {
+        let store_path = format!("{}/store.db", base_path);
+        let store = Store::persistent(&store_path).expect("Failed to create persistent store");
+        DWebBackend {
+            path: base_path.to_string(),
+            port,
+            store,
+            veilid_api: None,
+        }
+    }
+
+    // Updated start method to initialize both Store and Veilid
+    pub async fn start(&mut self) -> eyre::Result<()> {
+        println!("Starting on {} with port {}", self.path, self.port);
+
+        // Ensure base directory exists
+        let base_dir = &self.path;
+        fs::create_dir_all(base_dir).await.map_err(|e| {
+            eyre::eyre!("Failed to create base directory {}: {}", base_dir, e)
+        })?;
+
+        // Initialize Veilid
+        let update_callback: Arc<dyn Fn(VeilidUpdate) + Send + Sync> = Arc::new(|update| {
+            info!("Received update: {:?}", update);
+        });
+
+        let mut rng = StdRng::from_entropy();
+        let random_suffix: u16 = rng.gen_range(10000..60000);
+
+        // Create a VeilidConfigInner instance
+        let config_inner = VeilidConfigInner {
```

Comment:
Is there a persistent identifier created for our veilid node? If so we should document that fact.

Reply:
I'll might have to look deeper but to be sure I understand your question.

Would the node ID be the identifier you are referring to?

If so, we could retrieve it with something like this:
`veilid.config().unwrap().get_veilid_state().config.network.routing_table.node_id)`

if not, I will look into the config docs to find the relevant info

Reply:
Hmm, disregard that for now, we can just focus on keeping public keys for the dht persistent, the node id is irrelevant